### ISSUE TITLE
Sangria 2.0.0 planning and timeline

### ISSUE THREAD

**Issue by travisbrown** [sangria-graphql/sangria#463]
Last week we [recovered access](https://github.com/sangria-graphql/sangria/issues/446#issuecomment-546281588) to the `org.sangria-graphql` group and published a Sangria [2.0.0-M1](https://github.com/sangria-graphql-org/sangria/releases/tag/v2.0.0-M1) milestone release with support for Scala 2.13, together with releases for most of Sangria's marshalling and streaming modules. Since then we've also published releases with 2.13 support for [seven of the eight](https://github.com/sangria-graphql/sangria/issues/440#issuecomment-546398600) remaining modules (the final module, [sangria-rxscala](https://github.com/sangria-graphql-org/sangria-rxscala), is only waiting on [RxScala](http://reactivex.io/rxscala/) to add 2.13 support).

The next step is deciding on what still needs to be done for the 2.0.0 release. There are [several open pull requests](https://github.com/sangria-graphql/sangria/pulls) that still need review, prioritization, and migration to the new GitHub organization, and [dozens of open issues](https://github.com/sangria-graphql/sangria/issues). I've created a [2.0.0 milestone](https://github.com/sangria-graphql-org/sangria/milestone/1) in the new GitHub organization for tracking preparation for this release, and have [migrated](https://github.com/sangria-graphql-org/sangria/pull/5) one of Oleg's pull requests that seems like it should be included.

We'll also need to decide on an approximate timeline for the release. I'd propose that we aim to have the milestone list finalized by the end of next week (November 8), and to leave a couple of weeks for implementation and testing (possibly including a 2.0.0-M2 release) after that, with November 22 as a target date for the 2.0.0 release. That would give us a full month between the first milestone and the final release.

In the meantime it would be extremely helpful for adopters to try out [2.0.0-M1](https://github.com/sangria-graphql-org/sangria/releases/tag/v2.0.0-M1) and the accompanying module releases and to report back here with any feedback.

/cc @felipesilva @sachee @yanns

**Comment by vasantteja** [comment]
Hi, please let me know if you want help with any coding help. I am interested to contribute.

**Comment by jregistr** [comment]
I'd like to suggest we adopt Vs Code's pattern for planning/exposing the work that is in progress for the next planned release. The idea is to have a central place where we mention what is being considered/worked on for the next release. 
Take a look at Vs Code's [October IterationPlan](https://github.com/microsoft/vscode/issues/82194) for example. 
But following this, I'd imagine we'd come up with something below 👇. Alternatively, we could use Github projects for issue status and an issue for general discussion and updates.

I would be happy to get some thoughts on this.
---

This issue captures the work in the planned release `2.0.0`. The planned release date for this update is November 22.

**Legend for description markers.**

Mark | Description
----| -------
🛠 | Work in progress
⚠️ | Blocked task
🚧 | Deferred task for next iteration
💪 | Being considered but work has not started
🔴 | Missing issue reference
🔵 | More investigation required to remove uncertainty

### Scala Version Update

- [x] 🛠 Support Scala 2.13 https://github.com/sangria-graphql/sangria/issues/440

### Spec Update
- [ ] 🛠 Support repeatable directives. https://github.com/sangria-graphql/sangria/issues/409

---

#### Update: 04/11/2019 
Release candidate(s) are now available. 
- [Link for Sangria 2.0.0-RC-1](https://github.com/sangria-graphql/sangria/issues/463)
- [Link for Library X 2.0.0-RC-1](https://github.com/sangria-graphql/sangria/issues/463)

**Comment by dhoepelman** [comment]
> the final module, sangria-rxscala, is only waiting on RxScala to add 2.13 support

 [Scala 2.13 support is done](https://github.com/ReactiveX/RxScala/pull/243), but unfortunately not released. [I've opened an issue about this](https://github.com/ReactiveX/RxScala/issues/244), hopefully a release can be made or I can get rights to do it.

**Comment by travisbrown** [comment]
@jregistr I'm personally reluctant to add a lot of new process here, primarily because I'm not sure anyone has capacity to own it, but also because I don't think there's much risk of duplicating work now that the initial round of 2.13 updates is done.

In my view our main goal should be to make sure that people who have opened (or reviewed) PRs or issues are aware of the move and the timeline and can advocate for the changes if they're still interested. If someone wants to volunteer to help with more formal coordination, though, we should definitely discuss it!

**Comment by yanns** [comment]
I tried to release the version `2.0.0-M2`.
But since the introduction of sbt modules, I cannot release artifacts for all scala versions.
I tried several hacks, like https://github.com/sangria-graphql-org/sangria/commit/1e0131bedbcc07652c2cc39837bb94c77d3600e4, without any success.

`sbt release` creates artifacts for the scala 2.13 version only.

**Comment by meier-christoph** [comment]
not sure if this may help here but we had issues when adding support for 2.13 to some multi module projects because no scala version could be used for all modules i.e. the standard crossScalaVersions could not be used.
We used the sbt-projectmatrix plugin which solved that issue. Its still experimental but it worked for us.
https://github.com/sbt/sbt-projectmatrix

**Comment by felixbr** [comment]
@yanns I don't know much about `sbt-release` specifically, but usually you have to prefix a command with a `+` to tell sbt to execute it for all versions specificed in `crossScalaVersions`.

e.g. `sbt +publishLocal`

**Comment by yanns** [comment]
@meier-christoph  I'm not sure if you are reporting an issue, or just a hint for other people

I'm using sangria with scala 2.12 & 2.13 projects without any issue.
All artifacts are cross-published: https://repo1.maven.org/maven2/org/sangria-graphql/

@felixbr `sbt release` does it automatically: https://github.com/sangria-graphql-org/sangria/blob/master/build.sbt#L110

**Comment by meier-christoph** [comment]
@yanns not an issue just sharing our solution on a different project, seems you already solved it anyway

**Comment by yanns** [comment]
I'm very happy to share with you the first RC for the 2.0.0 release.
https://github.com/sangria-graphql-org/sangria/releases/tag/v2.0.0-RC1

If we have no severe issues with it, it will become the 2.0.0 release.

**Comment by btmcnellis** [comment]
@yanns What's the status of the 2.0.0 release? Are there any blockers identified in RC1 that people can help with?

**Comment by yanns** [comment]
Release 2.0.0 is out 🥳
https://github.com/sangria-graphql/sangria/releases/tag/v2.0.0

**Comment by btmcnellis** [comment]
Great news! Thanks to everyone who helped.
